sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql

```diff
@@ -418,8 +424,15 @@ android_iap_subscriptions AS (
     canceled_for_customer_at,
     CAST(NULL AS TIMESTAMP) AS cancel_at,
     CAST(NULL AS BOOL) AS cancel_at_period_end,
-    IF(end_time < TIMESTAMP(CURRENT_DATE), end_time, NULL) AS ended_at,
-    LEAST(end_time, TIMESTAMP(CURRENT_DATE)) AS end_date,
+    IF(
+      in_billing_grace_period,
+      -- android subscriptions in grace period have not ended
+      STRUCT(CAST(NULL AS TIMESTAMP) AS ended_at, TIMESTAMP(CURRENT_DATE) AS end_date),
```

Comment:
Note that I'm not suggesting we change how `end_date` works in this PR, as it looks like that's exposed in the final table+view and so any changes to it could have downstream impacts.